app/search/dataone.py

```diff
@@ -132,43 +132,40 @@ def convert_result(self, result):
             end = datetime.fromisoformat(result.pop('endDate').rstrip('Z'))
             result['temporal_coverage'] = datetime.date(
                 begin).isoformat() + "/" + datetime.date(end).isoformat()
+
         boundingbox = {'south': result.pop('southBoundCoord', None), 'north': result.pop(
             'northBoundCoord', None), 'west': result.pop('westBoundCoord', None), 'east': result.pop('eastBoundCoord', None)}
 
         if boundingbox:
             # Make a Point if the (north and south) and (east and west) have the same coordinates
             if boundingbox['north'] == boundingbox['south'] and boundingbox['east'] == boundingbox['west']:
-                geometry = Point(coordinates=(boundingbox['north'], boundingbox['east'])
+                bbox = Point(coordinates=(boundingbox['east'], boundingbox['north'])
                                  )
             else:
-                # Make a polygon with points in a counter clockwise motion and close the polygon by ending with the starting point
-                geometry = Polygon(
-                    coordinates=[
-                        [(boundingbox['east'], boundingbox['south']),
-                         (boundingbox['east'], boundingbox['north']),
-                            (boundingbox['west'], boundingbox['north']),
-                            (boundingbox['west'], boundingbox['south']),
-                            (boundingbox['east'], boundingbox['south']), ]
-                    ]
-                )
+                bbox = SearchResult.polygon_from_box(boundingbox)
+
+        geometry = {
+            # Named places, in DataONE results
+            'text': result.pop('placeKey', None),
+            'geometry_collection': bbox
+        }
         # passing the dictionary with the original data sources
-        self.data_source_key =  result.pop('datasource', '').lstrip("urn:node:")
+        self.data_source_key = result.pop('datasource', '').lstrip("urn:node:")
         if self.data_source_key in helper.get_original_dataone_sources():
-            datasource = helper.get_original_dataone_sources()[self.data_source_key]
-        
-            
-
+            datasource = helper.get_original_dataone_sources()[
+                self.data_source_key]
 
+        # Assure that author is a list
```

Comment:
The result template iterates through `author` to display each element in the list, but if there's only a single author that is a string, it iterates through the string instead and shows it like `M e l i n d a`